src/Tools/dotnet-counters/CounterPayload.cs

```diff
@@ -8,102 +8,95 @@
 
 namespace Microsoft.Diagnostics.Tools.Counters
 {
-    public interface ICounterPayload
+    public class CounterPayload
     {
-        string GetName();
-        double GetValue();
-        string GetDisplay();
-        string GetCounterType();
-    }
-
-
-    class CounterPayload : ICounterPayload
-    {
-        public string m_Name;
-        public double m_Value;
-        public string m_DisplayName;
-        public string m_DisplayUnits;
-
-        public CounterPayload(IDictionary<string, object> payloadFields)
+        public CounterPayload(string providerName, string name, string displayName, string displayUnits, string tags, double value, DateTime timestamp, string type)
         {
-            m_Name = payloadFields["Name"].ToString();
-            m_Value = (double)payloadFields["Mean"];
-            m_DisplayName = payloadFields["DisplayName"].ToString();
-            m_DisplayUnits = payloadFields["DisplayUnits"].ToString();
-
-            // In case these properties are not provided, set them to appropriate values.
-            m_DisplayName = m_DisplayName.Length == 0 ? m_Name : m_DisplayName;
+            ProviderName = providerName;
+            Name = name;
+            Tags = tags;
```

Comment:
Tags are labels/dimensions right? Is it fairly easy to convert these to a set of values instead of a single value, since we would need to do this conversion for dotnet-monitor?

Reply:
> Tags are labels/dimensions right?

Yep!

> Is it fairly easy to convert these to a set of values instead of a single value

Yeah it is a comma separated list of `<key>=<value>,<key2>=<value2>, ...`
Serializing it as a flat string wasn't my first choice, but until we do a better job with EventSource support for KeyValuePair[], string appeared to be the best alternative.